src/ActionList/Item.tsx

```diff
@@ -84,12 +84,21 @@ export const Item = React.forwardRef<HTMLLIElement, ActionListItemProps>(
       },
     }
 
+    const isTopLevelInteractive = () =>
```

Comment:
Should we use this to throw an error (or a warning) if somebody tries to render an interactive element as a child of an ActionList.Item who's top level is interactive?

Reply:
I don't think so. There isn't a good way to check the parent component without passing a ref. And in general, it leads to a weird layout, so I don't think it should be much of an actual issue. Here's an example.

```
<ActionList>
  <ActionList.Item as="button">
    <ActionList.Item as="button">Testing</ActionList.Item>
  </ActionList.Item>
</ActionList>
```

And it ends up looking like this.
<img width="1724" alt="Screenshot 2023-06-06 at 10 00 30 AM" src="https://github.com/primer/react/assets/1825422/9b64f785-e128-47c5-8113-849e2ae60937">